semantickernel-experimental/src/main/java/com/microsoft/semantickernel/data/record/options/VectorSearchOptions.java

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.data.record.options;
+
+import com.microsoft.semantickernel.builders.SemanticKernelBuilder;
+import com.microsoft.semantickernel.data.vectorsearch.filtering.BasicVectorSearchFilter;
+
+import javax.annotation.Nullable;
+
+public class VectorSearchOptions {
+
+    public static final int DEFAULT_RESULT_LIMIT = 3;
+
+    /**
+     * Creates a new instance of the VectorSearchOptions class with default values.
+     *
+     * @param vectorFieldName The name of the vector field.
+     * @return A new instance of the VectorSearchOptions class with default values.
+     */
+    public static VectorSearchOptions createDefault(String vectorFieldName) {
+        return VectorSearchOptions.builder()
+            .withVectorFieldName(vectorFieldName).build();
+    }
+
+    @Nullable
+    private final BasicVectorSearchFilter basicVectorSearchFilter;
+    @Nullable
+    private final String vectorFieldName;
+    private final int limit;
+    private final int offset;
+    private final boolean includeVectors;
+
+    public VectorSearchOptions(BasicVectorSearchFilter basicVectorSearchFilter,
+        String vectorFieldName, int limit, int offset, boolean includeVectors) {
+        this.basicVectorSearchFilter = basicVectorSearchFilter;
+        this.vectorFieldName = vectorFieldName;
+        this.limit = limit;
```

Comment:
Do limit or offset need bounds?

Reply:
Added verification to builder.

Reply:
Better to clamp the values in the constructor. I can still call the constructor with limit or offset out of bounds.